torch/hub.py

```diff
@@ -136,6 +136,16 @@ def _parse_repo_info(github):
                 ref = 'master'
             else:
                 raise
+        except OSError as e:  # Accounting for timeout error such that the code loads from cache when there is no internet
+            ref = None
+            for possible_ref in ['main', 'master']:
+                is_ref_in_cache = os.path.exists(
+                    f"{get_dir()}/{repo_owner}_{repo_name}_{possible_ref}")
+                if is_ref_in_cache:
+                    ref = possible_ref
+                    break
+            if ref == None:
+                raise
```

Comment:
Code looks good, although I'd suggest to fail with an informative error message if the `ref` can't be found in the cache (also simplifying a few unnecessary variables):

```suggestion
        except URLError as e:
            # No internet connection, need to check for cache as last resort
            for possible_ref in ("main", "master"):
                if os.path.exists(f"{get_dir()}/{repo_owner}_{repo_name}_{possible_ref}"):
                    ref = possible_ref
                    break
            if ref is None:
                raise RuntimeError(
                    "It looks like there is no internet connection and the "
                    f"repo could not be found in the cache ({get_dir()})"
                ) from e
 
```

Reply:
Yes, you are right. However, I had to import URLError, maybe include that in your suggestion as well? **Line 15** of `pytorch:master/torch/hub.py` should be  of  should be: 

```
from urllib.error import HTTPError, URLError
```
Thanks

Reply:
Committed above suggestions, please review.